### ISSUE TITLE
Dask backend returns early if no workers allocated

### ISSUE THREAD

**Issue by jcrist** [joblib/joblib#828]
If the cluster has no workers, the dask backend returns immediately.

```python
In [1]: from dask.distributed import Client, LocalCluster

In [2]: cluster = LocalCluster(n_workers=0)

In [3]: client = Client(cluster)

In [4]: import joblib

In [5]: with joblib.parallel_backend('dask'):
   ...:     results = joblib.Parallel(verbose=100)(joblib.delayed(lambda x: x**2)(x) for x in range(10))
   ...:
[Parallel(n_jobs=-1)]: Using backend DaskDistributedBackend with 0 concurrent workers.
[Parallel(n_jobs=-1)]: Done   0 out of   0 | elapsed:    0.0s finished

In [6]: import dask

In [7]: joblib.__version__
Out[7]: '0.13.0'

In [8]: dask.__version__
Out[8]: '1.0.0'
```

In this case the backend should probably hang until a worker is added, or error immediately. This is complicated since the number of workers could change dynamically during computation. I'm not sure what's best here.

xref https://github.com/dask/dask-yarn/issues/46

**Comment by jcrist** [comment]
cc @TomAugspurger for thoughts

**Comment by TomAugspurger** [comment]
Agreed that this should block until there are workers or error immediately. Of the two, I think I would prefer erroring, so that the following raises

```python
cluster = LocalCluster(n_workers=0)
client = Client(cluster)

with job lib.parallel_backend('dask'):  # raises here
    pass
```

but I'm relatively unsure of that preference.

**Comment by ogrisel** [comment]
I just encountered this bug myself today on a script with dask-jobqueue...

Maybe erroring by default is a good idea but we should also have an option to be able to wait for the first worker to be provisioned.

**Comment by ogrisel** [comment]
> Or would execution be handed off to the backend, and it would be responsible for polling for workers and timing out if needed?

The backend API only exposes the `effective_n_jobs` counter. At the moment it is never refreshed once a parallel call has started. I think advanced worker management logic should not be left to the internals of the backend implementation.

**Comment by ogrisel** [comment]
For the record, this issue is now fixed in joblib 0.13.1.
